server/src/main/java/com/flipkart/varadhi/auth/DefaultAuthorizationProvider.java

```diff
@@ -102,6 +102,9 @@ private List<Pair<ResourceType, ResourceContext>> generateResourceContextHierarc
             return List.of();
         }
 
+        // trim leading and trailing slashes
```

Comment:
Resource Path as a leading `/` which causes blank segment at index 0.
For some cases, there also comes a trailing `/` in case of create topic or create sub.
Apart from that it's just a measure to ensure input is sanitised.